Source/Diagnostics/ReducedDiags/FieldReduction.cpp

```diff
@@ -41,6 +43,10 @@ FieldReduction::FieldReduction (std::string rd_name)
     m_parser = std::make_unique<ParserWrapper<m_nvars>>(
         makeParser(parser_string,{"x","y","z","Ex","Ey","Ez","Bx","By","Bz"}));
 
+    // Replace all newlines and possible following whitespaces with a single whitespace. This
+    // should avoid weird formatting when the string is written in the header of the output file.
+    parser_string = std::regex_replace(parser_string, std::regex(("\n\\s*")), " ");
```

Comment:
Do we need to apply this transformation generally @WeiqunZhang ?

In that case, we might want to add it to the `Store_parserString` function in `Source/Utils/WarpXUtil.H`?

Reply:
Is there a parenthesis too much?
```suggestion
    parser_string = std::regex_replace(parser_string, std::regex((\n\\s*"), " ");
```

Reply:
I think the current regex does not work if a user starts at the beginning of the next line.
Maybe instead this?
```suggestion
    parser_string = std::regex_replace(parser_string, std::regex("\n"), " ");
```

Reply:
Hi @ax3l 

> Why do multi-line parser functions work for all other input at the moment? I use this a lot for density profiles.

Sorry I did not really explain in the PR description. The parser itself always work with multiple lines. However, in this specific case, we write the parser string in the header of the output text file. If there is a newline in the string, we end up with a multiline header that breaks `np.genfromtext()` in the analysis routines. This can be slightly inconvenient.

>  Do we need to apply this transformation generally?
In that case, we might want to add it to the Store_parserString function in Source/Utils/WarpXUtil.H?

I think that for now we only need it here. I don't exclude that we might need it at another place at some point, in which case it might make sense to move it to `Store_parserString`. As you prefer, I'm fine with both.

> Is there a parenthesis too much?

Yes, thanks I'll commit that.

> I think the current regex does not work if a user starts at the beginning of the next line.
Maybe instead this?

I've tested and it looks like it's working even if the string starts with a newline. Your suggestion would not remove the whitespaces immediatly after a newline. This should work (as far as `np.genfromtext()` is concerned) but the headers could potentially end up being ugly with many whitespaces where the newline was.

Reply:
Thanks for the details and updating the PR description!

> I've tested and it looks like it's working even if the string starts with a newline

Ah right. Actually `*` means "0 or more", that was my misunderstanding.